src/Development/IDE/Core/Compile.hs

```diff
@@ -122,7 +123,7 @@ compileModule
     :: HscEnv
     -> [TcModuleResult]
     -> TcModuleResult
-    -> IO ([FileDiagnostic], Maybe CoreModule)
+    -> IO ([FileDiagnostic], Maybe (CoreModule, TcModuleResult))
```

Comment:
We use this function and the `GenerateCore` rule in DAML where generating bytecode doesn’t make sense (if it works at all) and would slow things down unnecessarily. I think the most sensible option here is to split this in two:

1. GenerateCore can be changed to return `CgGuts`.
2. We can add a new `GenerateByteCode` rule that depends on `GenerateCore` to get the `CgGuts` and then returns `(CompiledByteCode, CgEntry)`.

I would probably just move the assembling of the `HomeModInfo` to the call site of this rule.

If you are fine with it, I’m also happy to do this change myself and push to your branch but if you want to give it a shot yourself you are more than welcome to!

Reply:
I would prefer you to do it, because I might not get it 100% right. I'll give you push access in my repo so you can just push whenever you are done.

Reply:
Great, I’ll take care of it (hopefully tomorrow but definitely this week). I don’t think you need to give me access to your repo. The default on github is that you can push to the branch of a PR to your repo so unless you disabled that I should be all set.